packages/patternfly-4/react-core/src/components/OptionsMenu/OptionsMenu.js

```diff
@@ -0,0 +1,91 @@
+import React, { Children, cloneElement, Component } from 'react';
+import PropTypes from 'prop-types';
+import styles from '@patternfly/patternfly/components/OptionsMenu/options-menu.css';
+import { css, getModifier } from '@patternfly/react-styles';
+import { OptionsMenuDirection, OptionsMenuPosition } from './optionsMenuConstants';
+
+const propTypes = {
+  /** Classes applied to root element of the Options menu */
+  className: PropTypes.string,
+  /** Id of the root element of the Options menu */
+  id: PropTypes.string.isRequired,
+  /** Array of OptionsMenuItem and/or OptionMenuItemGroup nodes that will be rendered in the Options menu list */
+  menuItems: PropTypes.arrayOf(PropTypes.node).isRequired,
+  /** Either an OptionsMenuToggleButton or a custom button to use to toggle the Options menu */
+  toggle: PropTypes.node.isRequired,
+  /** Provides an accessible name for the Options menu */
+  title: PropTypes.string,
+  /** Flag to indicate if menu is open */
+  isOpen: PropTypes.bool,
+  /** Flag to indicate the toggle has no border or background */
+  isPlain: PropTypes.bool,
+  /** For use when the .pf-c-options-menu__toggle is a <div> or some non-interactive elment, and you're using a custom .pf-c-options-menu__toggle-button to toggle the options menu. */
+  isText: PropTypes.bool,
+  /** Display menu above or below Options menu toggle */
+  direction: PropTypes.oneOf(Object.values(OptionsMenuDirection)),
+  /** Indicates where menu will be aligned horizontally */
+  position: PropTypes.oneOf(Object.values(OptionsMenuPosition)),
+};
+
+const defaultProps = {
+  className: '',
+  title: '',
+  isOpen: false,
+  isPlain: false,
+  isText: false,
+  direction: OptionsMenuDirection.down,
+  position: OptionsMenuPosition.left,
+};
+
+class OptionsMenu extends Component {
+
+  render() {
+    const {
+      className,
+      direction,
+      position,
+      id,
+      isOpen,
+      isPlain,
+      isText,
+      title,
+      menuItems,
+      toggle } = this.props;
+    return (
+      <div id={id}
+        className={
+          css(styles.optionsMenu,
+            direction === OptionsMenuDirection.up && getModifier(styles, 'top'),
+            position === OptionsMenuPosition.right && getModifier(styles, 'align-right'),
+            isOpen && getModifier(styles, 'expanded'),
+            className)}>
+        <span id={`${id}-label`} hidden>{title || id}</span>
+        {isText ? (
```

Comment:
I think we want to have a separate component here that applies the styles associated with Plain with Text toggle.  If we do it the way it is implemented here, we are making assumptions about the way the consumer implemented the toggle.   We are also depending on the consumer to apply the correct styles to their button and text span.